src/torchmetrics/wrappers/multitask.py

```diff
@@ -0,0 +1,110 @@
+from typing import Any, Dict, Optional, Sequence, Union
+
+from torch import Tensor
+
+from torchmetrics import Metric, MetricCollection
+from torchmetrics.utilities.imports import _MATPLOTLIB_AVAILABLE
+from torchmetrics.utilities.plot import _AX_TYPE, _PLOT_OUT_TYPE
+
+if not _MATPLOTLIB_AVAILABLE:
+    __doctest_skip__ = ["MultitaskWrapper.plot"]
+
+
+class MultitaskWrapper(Metric):
+    """Wrapper class for computing different metrics on different tasks in the context of multitask learning.
+
+    It computes a different metric on each pred and target given to its `update` method.
+
+    Args:
+        task_metrics:
+            Dictionary associating each task to a Metric or a MetricCollection. The keys of the dictionary represent the
+            names of the tasks, and the values represent the metrics to use for each task.
+
+    Example:
+         >>> import torch
+         >>> from torchmetrics.wrappers import MultitaskWrapper
+         >>> from torchmetrics.regression import MeanSquaredError
+         >>> from torchmetrics.classification import BinaryAccuracy
+         >>> classification_target = torch.tensor([0, 1, 0])
+         >>> regression_target = torch.tensor([2.5, 5.0, 4.0])
+         >>> targets = {"Classification": classification_target, "Regression": regression_target}
+         >>> classification_preds = torch.tensor([0, 0, 1])
+         >>> regression_preds = torch.tensor([3.0, 5.0, 2.5])
+         >>> preds = {"Classification": classification_preds, "Regression": regression_preds}
+         >>> metrics = MultitaskWrapper({
+         ...     "Classification": BinaryAccuracy(),
+         ...     "Regression": MeanSquaredError()
+         ... })
+         >>> metrics.update(preds, targets)
+         >>> metrics.compute()
+         {'Classification': tensor(0.3333), 'Regression': tensor(0.8333)}
+    """
+
+    is_differentiable = False
+
+    def __init__(
+        self,
+        task_metrics: Dict[str, Union[Metric, MetricCollection]],
+    ) -> None:
+        self._check_task_metrics_type(task_metrics)
+        super().__init__()
+        self.task_metrics = task_metrics
+
+    @staticmethod
+    def _check_task_metrics_type(task_metrics: Dict[str, Union[Metric, MetricCollection]]) -> None:
+        if not isinstance(task_metrics, dict):
+            raise TypeError(f"Expected task_metrics to be a dict. Found task_metrics = {task_metrics}")
+
+        for metric in task_metrics.values():
+            if not (isinstance(metric, (Metric, MetricCollection))):
+                raise TypeError(
+                    "Expected each task's metric to be a Metric or a MetricCollection. "
+                    f"Found a metric of type {type(metric)}"
+                )
+
+    def update(self, task_preds: Dict[str, Tensor], task_targets: Dict[str, Tensor]) -> None:
```

Comment:
```suggestion
    def update(self, *args: Tuple[Dict[str, Tensor]], **kwargs: Dict[str, Dict[str, Tensor]) -> None:
```
I wonder if we can generalize to `*args, **kwargs`. The wast majority of all torchmetrics metrics follows the `preds, target` naming, but a few does not. If we could do this, it would be cool, but if not I am okay with that.

Reply:
I don't understand the signature that you proposed, could you please explain it?

From my understanding, if we want to support some metrics which take arbitrary arguments in their `update` method, we need the `update` method of `MultitaskWrapper` to have the following signature:
```python
def update(self, task_positional_args: [Dict[str, Any]], task_kwargs: Dict[str, Dict[str, Any]) -> None:
```
i.e. we allow some arbitrary positional args and key-word args, different for each task.
I think a problem with this would be the use-case. We want the users to be able to do:
```python
preds = model(x)  # preds is of the form {"Classification": tensor(...), "Regression": tensor(...)}
multitask_metric.update(preds, targets)
```
with the above signature, we would have to do:
```python
preds = model(x)  # preds is of the form {"Classification": tensor(...), "Regression": tensor(...)}
positional_args = {
    "Classification": [preds["Classification"], targets["Classification"]], 
    "Regression": [preds["Regression"], targets["Regression"]]
}
multitask_metric.update(positional_args, {})
```
This makes this wrapper have a different usage than single-task metrics, which is precisely what I want to avoid by implementing the multi-task metric (so that we can change between single-task and multi-task problems without changing the code).

Again, I'm not sure that I understood what you meant, so please correct me if you had a different idea in mind!

Reply:
Cool, then lets just keep it as it is, but then lets specify in the documentation docstring that it is only metrics with the `preds, target` signature that is supported